libs/common/src/sql-db-seeder/utilisation-report/utilisation-report.seed.ts

```diff
@@ -0,0 +1,53 @@
+import { Seeder } from 'typeorm-extension';
+import { DataSource } from 'typeorm';
+import { UtilisationReportEntity } from '../../sql-db-entities';
+import { createNotReceivedReport, createMarkedAsCompletedReport, createUploadedReport } from './utilisation-report.helper';
+import { getCurrentReportPeriodForBankSchedule } from '../../helpers/utilisation-reports';
+import { MOCK_BANKS } from '../../test-helpers/mock-data';
+import { getUsersFromMongoDb } from '../helpers';
+
+export default class UtilisationReportSeeder implements Seeder {
+  /**
+   * Track seeder execution.
+   *
+   * Default: true
+   */
+  track = true;
+
+  public async run(dataSource: DataSource): Promise<void> {
+    const { paymentReportOfficer, pdcReconcileUser } = await getUsersFromMongoDb();
+
+    const utilisationReportRepository = dataSource.getRepository(UtilisationReportEntity);
+
+    let createdUploadedReport = false;
+    let createdMarkedAsCompletedReport = false;
+    const reportsToInsert = MOCK_BANKS.filter((bank) => bank.isVisibleInTfmUtilisationReports).map((bank) => {
+      const { id: bankId, utilisationReportPeriodSchedule } = bank;
+      const reportPeriod = getCurrentReportPeriodForBankSchedule(utilisationReportPeriodSchedule);
+
+      if (paymentReportOfficer.bank.id === bankId && !createdUploadedReport) {
```

Comment:
We no longer have access to `MOCK_BANKS`, though I agree that would have been better

Reply:
From an attempt at writing it this way, I actually think the loop is much more readable than trying to specifically select each bank to create the report for. The code becomes significantly more bloated if you do it another way as
- you need to write some function which finds the bank you want (for the payment report officer) and then pops that bank using `.splice` which I don't like because that function would have a side effect (you could return an object with the found bank and the remaining banks, but that would again result in more bloated code)
- you need to abstract away the extraction of the `bankId`, `utilisationReportPeriodSchedule` and generation of `reportPeriod` to another function so that you're not repeating it in the `run` function

I think the loop communicates what we're trying to do clearly enough, but let me know if you want to quickly pair on this to see if we can come up with something nicer

Reply:
Yeah let's chat offline, part of it is me wanting to have specific banks with certain reports in states so it's the same each time it runs even if we make any changes to the list of banks etc.